Sources/Capabilities/Filters/Views/MediaEditorFiltersViewController.swift

```diff
@@ -0,0 +1,170 @@
+            MediaEditorFilter(
+                name: "Mono",
+                ciFilterName: "CIPhotoEffectMono"
+            ),
+            MediaEditorFilter(
+                name: "Noir",
+                ciFilterName: "CIPhotoEffectNoir"
+            ),
+            MediaEditorFilter(
+                name: "Vintage",
+                ciFilterName: "CIPhotoEffectProcess"
+            ),
+            MediaEditorFilter(
+                name: "Tonal",
+                ciFilterName: "CIPhotoEffectTonal"
+            ),
+            MediaEditorFilter(
+                name: "Transfer",
+                ciFilterName: "CIPhotoEffectTransfer"
+            ),
+            MediaEditorFilter(
+                name: "Chrome",
+                ciFilterName: "CIPhotoEffectChrome"
+            ),
+            MediaEditorFilter(
+                name: "Fade",
+                ciFilterName: "CIPhotoEffectFade"
+            ),
+            MediaEditorFilter(
+                name: "Instant",
+                ciFilterName: "CIPhotoEffectInstant"
+            )
+        ]
+    }
+
+    let context = MediaEditor.ciContext
+
+    var onFinishEditing: ((UIImage, [MediaEditorOperation]) -> ())?
+
+    var onCancel: (() -> ())?
+
+    private var selectedFilterIndex = IndexPath(row: 0, section: 0)
+
+    override func viewDidLoad() {
+        imageView.image = image
+        filtersCollectionView.dataSource = self
+        filtersCollectionView.delegate = self
+    }
+
+    @IBAction func cancel(_ sender: Any) {
+        onCancel?()
+    }
+
+    @IBAction func done(_ sender: Any) {
+        guard selectedFilterIndex.row > 0 else {
+            onCancel?()
+            return
+        }
+
+        guard let ciImage = imageView.image?.ciImage, let cgImage = context.createCGImage(ciImage, from: ciImage.extent) else {
+            onCancel?()
+            return
+        }
+
+        onFinishEditing?(UIImage(cgImage: cgImage), [.filter])
+    }
+
+    func sepiaFilter(_ input: CIImage, intensity: Double) -> CIImage? {
+        let sepiaFilter = CIFilter(name:"CISepiaTone")
+        sepiaFilter?.setValue(input, forKey: kCIInputImageKey)
+        sepiaFilter?.setValue(intensity, forKey: kCIInputIntensityKey)
+        return sepiaFilter?.outputImage
+    }
+
+    func filter(_ image: UIImage, name: String) -> UIImage {
+        guard let ciImage = CIImage(image: image) else {
+            return image
+        }
+
+        let sepiaFilter = CIFilter(name: name)
+        sepiaFilter?.setValue(ciImage, forKey: kCIInputImageKey)
+
+        guard let outputImage = sepiaFilter?.outputImage else {
+            return image
+        }
+        return UIImage(ciImage: outputImage)
+    }
+
+    static func initialize() -> MediaEditorFiltersViewController {
+        return UIStoryboard(name: "MediaEditorFilters", bundle: Bundle(for: MediaEditorFiltersViewController.self)).instantiateViewController(withIdentifier: "filtersViewController") as! MediaEditorFiltersViewController
```

Comment:
This line is pretty long, what about breaking it up into multiple segments?

Reply:
Done in 59876f46698be33b72a234ffa6c9464c37854cde

What do you think?

Reply:
👍Works for me!